src/main/scala/diplomacy/package.scala

```diff
@@ -7,6 +7,168 @@ import chisel3.Data
 import freechips.rocketchip.config.Parameters
 import scala.language.implicitConversions
 
+
+/**
+  * Diplomacy is a set of abstractions for describing directed, acyclic graphs
+  * where parameters will be negotiated between nodes. These abstractions are
+  * expressed in the form of abstract classes, traits, and type parameters, which
+  * comprises nearly all of the types defined in this file.
+  *
+  * The [[NodeImp]] ("node implementation") is the main abstract type that associates
+  * the type parameters of all other abstract types. Defining a concrete
+  * implementation of [[NodeImp]] will therefore determine concrete types for all
+  * type parameters. For example, passing in a concrete instance of NodeImp to a
+  * SourceNode will fully determine concrete types for all of a SourceNode's type
+  * parameters.
+  *
+  * Specific applications of Diplomacy are expected to either extend these types
+  * or to specify concrete types for the type parameters. This allows for
+  * creating and associating application-specific node, edge, parameter, and bundle types.
+  *
+  *
+  * =Concepts, metaphors, and mnemonics to help with understanding Diplomacy code=
+  *
+  * ==Parameter Types==
+  *  
+  *  There are several types of parameters involved in diplomacy code.
+  *  
+  *  - Up/Down Parameters: These parameter types flow along edges and can be considered as the
+  *    pre-negotiated, unresolved parameters.
+  *  - Edge Paramters: These parameters are the result of the diplomatic negotiation and correspond
+  *    to an edge. These are an abstract concept which carry any sort of conceptual information that is useful to 
```

Comment:
`They are metadata, or an abstract concept of the connection represented by their edge, and contain any sort...`